.github/scripts/install-slices/install_slices.py

```diff
@@ -263,41 +282,59 @@ def ignore_missing_packages(
     return filtered, ignored
 
 
-def install_slice(
-    pkg: str, slice: str, arch: str, release: str, missing_copyright: set
+def install_slices(
+    chunk: list[tuple], dry_run: bool, arch: str, release: str, worker: int
 ) -> None:
     """
     Install the slice by running "chisel cut".
     """
-    slice_name = full_slice_name(pkg, slice)
-    logging.info("Installing %s on %s...", slice_name, arch)
-    with tempfile.TemporaryDirectory() as tmpfs:
-        res = subprocess.run(
-            args=[
-                "chisel",
-                "cut",
-                "--arch",
-                arch,
-                "--release",
-                release,
-                "--root",
-                tmpfs,
-                slice_name,
-            ],
-            capture_output=True,
-            text=True,
-            check=False,
+    for i, (pkg, slice) in enumerate(chunk):
+        slice_name = full_slice_name(pkg, slice)
+        logging.info(
+            "Worker %d (%d/%d): Installing %s on %s...",
+            worker,
+            i,
+            len(chunk),
+            slice_name,
+            arch,
         )
-        if res.returncode != 0:
-            logging.error(
-                "==============================================\n%s",
-                res.stderr.rstrip(),
+        if dry_run:
+            continue
+        with tempfile.TemporaryDirectory() as tmpfs, tempfile.TemporaryDirectory() as cache_dir:
+            os.environ["XDG_CACHE_HOME"] = str(cache_dir)
```

Comment:
maybe worth making a temp environ and passing that, instead of patching the current one? something like `env = dict(os.environ); env['XDG_CACHE_HOME'] = str(cache_dir)`. i don't think it matters much here since there is sys.exit right after, but otherwise it would patch the `os.environ` for the rest of the process